### PR TITLE
Add unit test for handle_link_trapped

### PR DESCRIPTION
**Pull request by MarkintoshZ**
Add the missing unit test for `handle_link_trapped`

### CODE REVIEW

**Comment by bkolobara** [comment]
This test is a bit hard to follow and reason about. I would simplify it to something like:

```
fn init(_process: ProcessRef<Self>, _arg: ()) -> Self {
    unsafe { host::api::process::die_when_link_dies(0) };
    spawn_link!(|| panic!());
    Self {
        link_trapped: false,
    }
}
```

Now you don't need a handler for the `Panic` message and you don't need to spawn two processes.

**Comment by MarkintoshZ** [comment]
Fixed!
